### PR TITLE
Issue #34 - space encoded as plus

### PR DESCRIPTION
**Pull request by homes2001**

### CODE REVIEW

File: src/main/java/vc/inreach/aws/request/AWSSigningRequestInterceptor.java [hunk 1]
```diff
@@ -36,7 +39,13 @@ public void process(HttpRequest request, HttpContext context) throws HttpExcepti
     }
 
     private Multimap<String, String> params(HttpRequest request) {
-        return params(((HttpRequestWrapper) request).getURI().getQuery());
+        try {
+            final String rawQuery = ((HttpRequestWrapper) request).getURI().getRawQuery();
```
Comment (thesmith) [inline src/main/java/vc/inreach/aws/request/AWSSigningRequestInterceptor.java]: This will often be null (and throw an NPE on the next line)

**Comment by thesmith** [comment]
Can this be squashed down or something? Not sure why there's the first commit...

**Comment by homes2001** [comment]
You are right, there is a NPE when there is no query string. I have added a test case to verify and to make sure this does not happen in the future anymore. Furthermore I have removed the first commit. Also I have removed the try/catch to make the code cleaner, as the calling function can cope with IOExceptions.

**Comment by homes2001** [comment]
@thesmith did you have a chance to review the new commits?

**Comment by thesmith** [comment]
All done!

**Pull request merged**